### PR TITLE
deal hooks

### PR DESCRIPTION
**Pull request by madalilng**
add deal hooks

### CODE REVIEW

File: src/Network/Receive.pm [hunk 1]
```diff
@@ -3027,6 +3027,7 @@ sub deal_begin {
 		error T("That person is too far from you to trade.\n"), "deal";
 	} elsif ($args->{type} == 2) {
 		error T("That person is in another deal.\n"), "deal";
+		Plugins::callHook("error_deal", { msg => "That person is in another deal."});
```
Comment (lututui) [inline src/Network/Receive.pm]: Since this is the only place error_deal is called, do we need a msg argument?

Comment (alisonrag) [reply src/Network/Receive.pm]: i think the message is not necessary or we need to add a message in others case like player is too far or using the storage

Comment (lututui) [reply src/Network/Receive.pm]: Passing $args{type} is better than passing a string as argument

Comment (alisonrag) [reply src/Network/Receive.pm]: so the better way is:
Plugins::callHook("error_deal", { type =>$args->{type}});

Comment (madalilng) [reply src/Network/Receive.pm]: will i remove the engaged_deal hook and add 
hooks at the bottom with { type =>$args->{type}} ??

Comment (lututui) [reply src/Network/Receive.pm]: You can keep the other hooks the way they are. Just change it for the error hooks

Comment (lututui) [reply src/Network/Receive.pm]: @Mortimal no, there are other codes different from 3 that are not errors

**Pull request merged**